### PR TITLE
Doc multinom

### PR DESCRIPTION
**Pull request by thibautjombart**
This adds a new vignette "discrete" on discrete deterministic and stochastic models, with a strong emphasis on the latter. Explains how binomial, Poisson, and multinomial distributions are used in this context.

Provides a simple example using SIR models and then a more advanced one using a SEIRDS model with imported cases. Still needs some proof-reading but should be useful as it is.

### CODE REVIEW

**Comment by richfitz** [comment]
Can you please re-point this at `i100_multinom` please?  If that's a faff don't worry - I can merge that one in at the same time.  I'll go through this tomorrow.  Thanks again!

**Comment by thibautjombart** [comment]
@richfitz sorry, didn't think about the base, no repointed to the right branch

File: vignettes/discrete.Rmd [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+x <- sir_model()
+x
+```
+
+`x` is an `ode_model` object which can be used to generate dynamics of a
+discrete-time, deterministic SIR model. This is achieved using the function
+`x$run()`, providing time steps as single argument, e.g.:
+
+``` {r sir-deterministic}
+sir_col <- c("#8c8cd9", "#cc0044", "#999966")
```
Comment (richfitz) [inline vignettes/discrete.Rmd]: My preference with the plots is to:

* drop the `main =` title
* set the margins to be smaller (`par(mar = c(4.1, 4.1, 0.5, 0.5))` or so)
* drop the box on the legend (`bty = "n"`)
* make the `lwd` just 1
* use `las = 1` to align the y labels more nicely

These are all negotiable, but eventually all the plots should have similar style across the package

File: vignettes/discrete.Rmd [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+x_res <- x$run(0:200)
+matplot(x_res[, 1], x_res[, -1], xlab = "Time", ylab = "Number of individuals",
+        main = "Discrete SIR model - deterministic", type = "l", col = sir_col,
+        lwd = 3, lty = 1)
+legend("topright", lwd = 3, col = sir_col, legend = c("S", "I", "R"))
+```
+
+
+
+Stochastic discrete SIR
```
Comment (richfitz) [inline vignettes/discrete.Rmd]: Missing title

File: vignettes/discrete.Rmd [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+
+The stochastic equivalent of the previous model can be formulated in `odin`
+as follows:
+
+``` {r load_sir_s}
+path_sir_model_s <- system.file("examples/discrete_stochastic_sir.R", package = "odin")
```
Comment (richfitz) [inline vignettes/discrete.Rmd]: can you please indicate which files you feel these included will have replaced the ones I popped into https://github.com/mrc-ide/odin/commit/9dfe154d41fa0c89f9df2e1abdd84ac89764bc9f and I'll rewire the tests to point your new versions

File: vignettes/discrete.Rmd [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+``` {r echo = FALSE, comment = NA}
+cat(readLines(path_sir_model_s), sep = "\n")
+```
+
+
+We can use the same workflow as before to run the model:
+``` {r }
+sir_model_s <- odin::odin(path_sir_model_s, verbose = FALSE, skip_cache = TRUE)
+sir_model_s
+x <- sir_model_s(I_ini = 10) # customise param: I_ini = 10 individuals
```
Comment (richfitz) [inline vignettes/discrete.Rmd]: I would lift the comment here into the prose

File: vignettes/discrete.Rmd [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+```
+
+``` {r echo = FALSE, comment = NA}
+cat(readLines(path_sir_model_s), sep = "\n")
+```
+
+
+We can use the same workflow as before to run the model:
+``` {r }
+sir_model_s <- odin::odin(path_sir_model_s, verbose = FALSE, skip_cache = TRUE)
```
Comment (richfitz) [inline vignettes/discrete.Rmd]: Elsewhere I call these "generators" (so `odin::odin` creates a _generator_, and running the _generator_ creates a _model_).  So perhaps re-name as:

```
sir_s <- odin::odin(...)
mod_sir_s <- sir_s(I_ini = 10)
```

there is no obviously correct set of names, and this does start making more sense as an approach once things are packaged

File: vignettes/discrete.Rmd [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+
+
+The formulation of the model in `odin` is:
+
+``` {r load_seirds}
+path_seirds_model <- system.file("examples/discrete_stochastic_seirds.R", package = "odin")
+```
+
+``` {r echo = FALSE, comment = NA}
+cat(readLines(path_seirds_model), sep = "\n")
```
Comment (richfitz) [inline vignettes/discrete.Rmd]: wrap these in `output_r` - see the source of the main vignette for details

File: vignettes/discrete.Rmd [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+```
+
+``` {r seirds}
+seirds_model <- odin::odin(path_seirds_model, verbose = FALSE, skip_cache = TRUE)
```
Comment (richfitz) [inline vignettes/discrete.Rmd]: drop the skip_cache

File: vignettes/discrete.Rmd (hunk 1 above)
Comment (thibautjombart) [reply vignettes/discrete.Rmd]: Will be done in revision.

File: vignettes/discrete.Rmd (hunk 2 above)
Comment (thibautjombart) [reply vignettes/discrete.Rmd]: Done at a6f247631b4c7478781632b642423fce1f37bd83

File: vignettes/discrete.Rmd (hunk 3 above)
Comment (thibautjombart) [reply vignettes/discrete.Rmd]: Corresponding test files are basically all files in `tests/testthat/stochastic`:
```
thibaut@nuxVM ~/dev/odin $ ls tests/testthat/stochastic/
sir_discrete.R  sir_discrete_stochastic2.R  sir_discrete_stochastic_multi.R  sir_discrete_stochastic.R
```

File: vignettes/discrete.Rmd (hunk 4 above)
Comment (thibautjombart) [reply vignettes/discrete.Rmd]: done at e1e05cf03417e32d3d9692bb2e857ae0b2814feb

File: vignettes/discrete.Rmd (hunk 5 above)
Comment (thibautjombart) [reply vignettes/discrete.Rmd]: Yes it is clearer this way: cc8886a4676f304410af4251e710c5ec191132e7

File: vignettes/discrete.Rmd (hunk 6 above)
Comment (thibautjombart) [reply vignettes/discrete.Rmd]: done at 9ce214493ddf2a86d75e3d01787f7e599817b0b0

File: vignettes/discrete.Rmd (hunk 7 above)
Comment (thibautjombart) [reply vignettes/discrete.Rmd]: done at 65299938ef7c1dc29349ba91984e3407a0062cb8

Does it mean it will be default behaviour?

**Comment by thibautjombart** [comment]
All changes done as of 65299938ef7c1dc29349ba91984e3407a0062cb8

**Pull request merged**